src/AgoraGovernor.sol

```diff
@@ -408,6 +414,30 @@ contract AgoraGovernor is
         return propose(targets, values, calldatas, description, 0);
     }
 
+    /**
+     * @notice Validates the proposed transactions against the defined scopes based on the proposal type
+     * proposal threshold can propose.
+     * @param targets The list of target contract addresses.
+     * @param calldatas The list of proposed transaction calldata.
+     * @param proposalType The type of the proposal.
+     */
+    function validateProposalData(address[] memory targets, bytes[] memory calldatas, uint8 proposalType)
```

Comment:
what if the length of `targets` and `calldatas` doesn't match?

Reply:
The check is still being performed in the `propose` function, we could move it here though.